security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,271 @@
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. end-to-end tests created for the L1 block deployment upgrade transaction
+      - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
+  2. same upgrade path as previous predeploys
+      - example: [EIP-4788 deployment transaction](https://github.com/ethereum-optimism/optimism/blob/b6b74290b5502f45daae57946f969327cdb2d383/op-node/rollup/derive/ecotone_upgrade_transactions.go#L130)
+    
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction in rollup node client software (like `op-node`).
+
+### FM2: Chain halt or fork due to request hash mismatch
+
+- **Description:** Requests hash was added to the block header. If it mismatches between EL clients, or the rollup node, the chain could halt or fork for the affected clients.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end test ensuring requests hash is always `sha256('')` indicating no requests are included in blocks.
+      - https://github.com/ethereum-optimism/optimism/pull/14253
+  2. Simplicity of validation rules - always must be an empty hash
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing requests hash logic in respective clients (e.g. `op-node`, `op-geth`, `op-reth`, ...)
+- **References**:
+    - [Appendix B](#appendix-b-block-header-changes) shows the changes to the block header.
+    - [specs: Header validity rules](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#header-validity-rules)
+    - [op-e2e: test ensuring empty hash even if requests created](https://github.com/ethereum-optimism/optimism/blob/9df1fc15d0bf0dc9464db249ce06424607d5f399/op-e2e/actions/proofs/isthmus_requests_test.go#L19)
+
+### FM3: Increased call data cost affects network upgrade transactions
+
+- **Description:** Increased calldata cost as the result of [EIP-7623](https://eips.ethereum.org/EIPS/eip-7623) could mean that network upgrade transactions fail.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Tested that network upgrade transactions don't fail.
+        - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
+  2. The EIP results in the same gas behavior as long as most of the gas is not spent on call data.
+        - Most upgrade transactions create a new contract, and contract creation gas remains the same.
```

Comment:
Also Kurtosis, the alpha/betanets and e2e tests all run through the upgrade block, so this would have been caught.